Bottom: 8c41982c881583c8ca7e21e2536b41d6471276a2
Top:    1ecb7a9b028b874199c0cce3a6549aa30ba44660
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-03-18 20:19:32 +0000

Improve dump_poly() output

Make it print each contour


---

diff --git a/src/polygon1.c b/src/polygon1.c
index a369b2d..7032370 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -144,20 +144,26 @@ pline_dump (VNODE * v)
 	       n->point[0], n->point[1], theState (v));
     }
   while ((v = v->next) != s);
-  fprintf (stderr, "NEXT PLINE\n");
 }
 
 static void
 poly_dump (POLYAREA * p)
 {
   POLYAREA *f = p;
+  PLINE *pl;
 
   do
     {
-      pline_dump (&p->contours->head);
+      pl = p->contours;
+      do
+        {
+          pline_dump (&pl->head);
+          fprintf (stderr, "NEXT PLINE\n");
+        }
+      while ((pl = pl->next) != NULL);
+      fprintf (stderr, "NEXT_POLY\n");
     }
   while ((p = p->f) != f);
-  fprintf (stderr, "NEXT_POLY\n");
 }
 #endif
